aws-cpp-sdk-transfer-tests/TransferTests.cpp

```diff
@@ -1322,5 +1324,106 @@ TEST_F(TransferTests, MultipartUploadWithMetadataTest)
     ASSERT_EQ(metadata["key1"], headObjectMetadata["key1"]);
     ASSERT_EQ(metadata["key2"], headObjectMetadata["key2"]);
 }
+    
+// Similar to SmallTest, but passes in a string stream for upload instead of a file name
+TEST_F(TransferTests, StreamSmallTest)
+{
+    if (EmptyBucket(GetTestBucketName()))
+    {
+        WaitForBucketToEmpty(GetTestBucketName());
+    }
+    
+    GetObjectRequest getObjectRequest;
+    getObjectRequest.SetBucket(GetTestBucketName());
+    getObjectRequest.SetKey(SMALL_FILE_KEY);
+    
+    GetObjectOutcome getObjectOutcome = m_s3Client->GetObject(getObjectRequest);
+    EXPECT_FALSE(getObjectOutcome.IsSuccess());
+    
+    ListMultipartUploadsRequest listMultipartRequest;
+    listMultipartRequest.SetBucket(GetTestBucketName());
+    
+    Aws::IFStream ifstream;
+    ifstream.open(SMALL_TEST_FILE_NAME);
+    ASSERT_TRUE(ifstream.is_open());
+    
+    auto sstream = Aws::MakeShared<Aws::StringStream>(ALLOCATION_TAG);
+    *sstream << ifstream.rdbuf();
+    ifstream.close();
+    
+    const bool cCreateBucket = false;
+    const bool cConsistencyChecks = false;
+    // Not creating the bucket.. it should be there
+    // No consistency checks on this one, just verifying that it seems to complete
+    std::shared_ptr<UploadFileRequest> requestPtr = m_transferClient->UploadFile(sstream, GetTestBucketName(), SMALL_FILE_KEY, "", cCreateBucket, cConsistencyChecks);
+    
+    ASSERT_EQ(requestPtr->GetTotalParts(), 1u); // Should be about 2.5 megs
+    
+    ASSERT_FALSE(requestPtr->IsDone());
```

Comment:
I don't think the upload waits for WaitForUploadAndUpdate, but I'm guessing the assumption is that the upload won't finish that quickly. I just followed the examples of the other tests for this.